### ISSUE TITLE
IdentifyTransactions message not implemented

### ISSUE THREAD

**Issue by lorbax** [stratum-mining/stratum#902]
IdentifyTransactions and .Succes are not implemented. The JDS just ignores the cases where it necessary to send this message (collisions with short hashes or impossible to reconstruct the `tx_hash_list_hash`). If the JDC receives such a message, [it responds with an empty vector](https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/lib/job_declarator/message_handler.rs#L45 t).

Good first issue

**Comment by rockcoolsaint** [comment]
@lorbax I don't quite understand what is to be done here.
What is the expected code deliverable?

What does this statement mean? `The JDS just ignores the cases where it necessary to send this message (collisions with short hashes or impossible to reconstruct the tx_hash_list_hash)`

**Comment by marathon-gary** [comment]
https://stratumprotocol.org/specification/06-Job-Declaration-Protocol/#617-identifytransactions-server-client
> Sent by the Server in response to a DeclareMiningJob message indicating it detected a collision in the `tx_short_hash_list`, or was unable to reconstruct the `tx_hash_list_hash`.

The client receives an IdentifyTransactions message from the server, indicating a collision or inability to reconstruct the hash list of tx's. Currently, there is [just an empty vec returned.](https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/lib/job_declarator/message_handler.rs#L45) 

According to spec it should be providing:
> The full list of transaction data hashes used to build the mining job in the corresponding DeclareMiningJob message

The original `DeclareMiningJob` only contains `tx_short_hash_list` and `tx_hash_list_hash`. An `IdentifyTransactions` message indicates there was a collision in the `short hash` list or the `hash list hash` couldn't be recreated.

I assume the client just looks up the original list of tx's and then sends it back in the `.Success` message.

I will take a pass at a PR for this

**Comment by lorbax** [comment]
> https://stratumprotocol.org/specification/06-Job-Declaration-Protocol/#617-identifytransactions-server-client
> 
> > Sent by the Server in response to a DeclareMiningJob message indicating it detected a collision in the `tx_short_hash_list`, or was unable to reconstruct the `tx_hash_list_hash`.
> 
> The client receives an IdentifyTransactions message from the server, indicating a collision or inability to reconstruct the hash list of tx's. Currently, there is [just an empty vec returned.](https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/lib/job_declarator/message_handler.rs#L45)
> 
> According to spec it should be providing:
> 
> > The full list of transaction data hashes used to build the mining job in the corresponding DeclareMiningJob message
> 
> The original `DeclareMiningJob` only contains `tx_short_hash_list` and `tx_hash_list_hash`. An `IdentifyTransactions` message indicates there was a collision in the `short hash` list or the `hash list hash` couldn't be recreated.
> 
> I assume the client just looks up the original list of tx's and then sends it back in the `.Success` message.
> 
> I will take a pass at a PR for this

The original list of `Txid` (32byes each) should be sent in the .Success, not just the short hashes (6bytes each) that are sent in DeclareMiningJob.
Or am I missing something about your comment?
Perhaps I am missing something about your comment?

Happy that someone finally decided to tackle this ^^

**Comment by marathon-gary** [comment]
> > https://stratumprotocol.org/specification/06-Job-Declaration-Protocol/#617-identifytransactions-server-client
> > > Sent by the Server in response to a DeclareMiningJob message indicating it detected a collision in the `tx_short_hash_list`, or was unable to reconstruct the `tx_hash_list_hash`.
> > 
> > 
> > The client receives an IdentifyTransactions message from the server, indicating a collision or inability to reconstruct the hash list of tx's. Currently, there is [just an empty vec returned.](https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/lib/job_declarator/message_handler.rs#L45)
> > According to spec it should be providing:
> > > The full list of transaction data hashes used to build the mining job in the corresponding DeclareMiningJob message
> > 
> > 
> > The original `DeclareMiningJob` only contains `tx_short_hash_list` and `tx_hash_list_hash`. An `IdentifyTransactions` message indicates there was a collision in the `short hash` list or the `hash list hash` couldn't be recreated.
> > I assume the client just looks up the original list of tx's and then sends it back in the `.Success` message.
> > I will take a pass at a PR for this
> 
> The list of `Txid` (32byes each) of declared job transactions should be sent in the .Success, not just the short hashes (6bytes each) that are sent in DeclareMiningJob. Or am I missing something about your comment?
> 
> Happy that someone finally decided to tackle this ^^

Yup! That's my understanding. Was mainly noting down my understanding for validation. Thnx for the Ack! :)

**Comment by CietyGoblin** [comment]
> https://stratumprotocol.org/specification/06-Job-Declaration-Protocol/#617-identifytransactions-server-client
> 
> > Sent by the Server in response to a DeclareMiningJob message indicating it detected a collision in the `tx_short_hash_list`, or was unable to reconstruct the `tx_hash_list_hash`.
> 
> The client receives an IdentifyTransactions message from the server, indicating a collision or inability to reconstruct the hash list of tx's. Currently, there is [just an empty vec returned.](https://github.com/stratum-mining/stratum/blob/dev/roles/jd-client/src/lib/job_declarator/message_handler.rs#L45)
> 
> According to spec it should be providing:
> 
> > The full list of transaction data hashes used to build the mining job in the corresponding DeclareMiningJob message
> 
> The original `DeclareMiningJob` only contains `tx_short_hash_list` and `tx_hash_list_hash`. An `IdentifyTransactions` message indicates there was a collision in the `short hash` list or the `hash list hash` couldn't be recreated.
> 
> I assume the client just looks up the original list of tx's and then sends it back in the `.Success` message.
> 
> I will take a pass at a PR for this

https://github.com/stratum-mining/stratum/issues/902#issuecomment-2214808996

**Comment by GitGab19** [comment]
Closing this as stale.

`IdentifyTransactions` message has been deprecated from the specs a while ago.
